mri/reconstruct/utils.py

```diff
@@ -188,7 +192,7 @@ def generate_operators(data, wavelet_name, samples, nb_scales=4,
             fourier_op=fourier_op)
 
     # Define the proximity dual/primal operator
-    prox_op = SparseThreshold(linear_op, None, thresh_type="soft")
+    prox_op = SparseThreshold(linear_op, mu, thresh_type="soft")
```

Comment:
It has to be at the right shape, so you need a vector not a float.

Reply:
I added the case where the hyper-parameter is an array

Reply:
I don't really understand why the my should be an array if the `SparseThreshold` works fine with a float, it is more efficient in terms of memory to have a float than an array

Reply:
no you are right it can be a float. I was just confuse because we had always been setting it with an array, but actually I checked and it can indeed by just a float, my bad.